src/class-last-modified-date.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * WP_Video_Sync: Last modified date.
+ *
+ * @package wp-video-sync
+ */
+
+namespace Alley\WP\WP_Video_Sync;
+
+use DateTimeImmutable;
+
+/**
+ * Class for tracking the modified date of the last video in a batch.
+ */
+class Last_Modified_Date {
+
+	/**
+	 * The date of the last modification to the last batch of videos.
+	 *
+	 * @var ?DateTimeImmutable
+	 */
+	private ?DateTimeImmutable $last_modified_date = null;
+
+	/**
+	 * Fetches the date of the last modification to the last batch of videos.
+	 *
+	 * @return ?DateTimeImmutable
+	 */
+	public function get_last_modified_date(): ?DateTimeImmutable {
+		return $this->last_modified_date;
+	}
+
+	/**
+	 * Sets the date of the last modification based on the latest batch of videos.
+	 *
+	 * @param string $last_modified_date The date of the last modified video in the batch.
+	 * @return void
+	 */
+	public function set_last_modified_date( string $last_modified_date = '' ): void {
```

Comment:
I feel pretty strongly that this should accept a `DateTimeImmutable` rather than a string. The date formats returned by different APIs won't necessarily be the same, so it's best to trade in a common format, which is `DateTimeImmutable`. This way, the JW Player API response (which returns an ISO-8601 date compatible with the `DATE_W3C` format of DateTime) can be converted to a DateTimeImmutable object in the adapter and passed here where it will be set. If another adapter gets dates in a different format (e.g., `Y-m-d H:i:s` or just `Y-m-d` or a Unix timestamp) that adapter would be responsible for standardizing on DateTimeImmtuable, etc.